benches/nixpkgs/lists.ncl

```diff
@@ -116,4 +116,554 @@
        >> [ "a", "z", "b", "z" ]
   "%m
   = (fun f list => array.flatten (array.map f list)),
+
+  flatten: Dyn -> Array Dyn
+  | doc m%"
+      Flatten the argument into a single list, that is, nested lists are
+     spliced into the top-level lists.
+
+     Example:
+       flatten [1, [2, [3], 4], 5]
+       >> [1, 2, 3, 4, 5]
+       flatten 1
+       >> [1]
+  "%m
+  = fun x =>
+    if builtin.is_array x
+    then concatMap (fun y => flatten y) (x | Array Dyn)
+    else [x],
+
+  remove: forall a. a -> Array a -> Array a
+  | doc m%"
+      Remove elements equal to 'e' from a list.  Useful for buildInputs.
+
+     Example:
+       remove 3 [ 1, 3, 4, 3 ]
+       >> [ 1, 4 ]
+  "%m
+  = # Element to remove from the list
+    fun e => array.filter (fun x => x != e),
```

Comment:
```suggestion
    fun e => array.filter ((!=) e),
```

Nitpick, but just to show how cool curried operator are :stuck_out_tongue: (and to test their typechecking too)